### PR TITLE
fix(PanoViewer): remove resume/suspend method

### PR DESCRIPTION
**Pull request by happyhj**
## Issue
#108

## Details
* remove resume/suspend method
* update unit tests
* update manual tests
* update examples
* update marked dependency version to avoid security vulnerability.
* added new error type `RENDERING_CONTEXT_LOST` triggering when webgl context lost occurring by browser.

### CODE REVIEW

File: src/PanoViewer/PanoViewer.js [hunk 1]
```diff
@@ -217,10 +204,6 @@ export default class PanoViewer extends Component {
 	}
 
 	_initRenderer(yaw, pitch, fov, projectionType) {
-		if (this._photoSphereRenderer) {
-			this._photoSphereRenderer.destroy();
-		}
-
```
[the remark this answers is not in the document]
Comment (jongmoon) [reply src/PanoViewer/PanoViewer.js]: @happyhj I found that _inactivateInteraction() is called before _initRenderer. So your remove is no problem.

File: src/PanoViewer/PanoViewer.js [hunk 2]
```diff
@@ -570,47 +554,26 @@ export default class PanoViewer extends Component {
...
+				this._triggerEvent(EVENTS.ERROR, {
+					type: ERROR_TYPE.FAIL_BIND_TEXTURE,
+					message: "failed to bind texture"
 				});
-		}, 0);
+			});
 	}
 
-	_resume() {
+	_activateInteraction() {
```
Comment (jongmoon) [inline src/PanoViewer/PanoViewer.js]: I think naming is narrow than it's functionality.

It activates interaction but also activates renderer.

File: src/PanoViewer/PanoViewer.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (happyhj) [reply src/PanoViewer/PanoViewer.js]: ok

File: src/PanoViewer/PanoViewer.js [hunk 3]
```diff
@@ -246,15 +240,17 @@ export default class PanoViewer extends Component {
...
-
-		this._photoSphereRenderer.on(PanoImageRenderer.EVENTS.RENDERING_CONTEXT_RESTORE, e => {
-			this.resume();
-		});
+			this._inactivateInteraction();
+			this.trigger(EVENTS.ERROR, {
+				type: ERROR_TYPE.RENDERING_CONTEXT_LOST,
+				message: "webgl rendering context lost"
+			});
+		}, false);
```
[the remark this answers is not in the document]
Comment (happyhj) [reply src/PanoViewer/PanoViewer.js]: It was mistake. I'll remove that.

File: src/PanoViewer/PanoViewer.js [hunk 4]
```diff
@@ -570,47 +554,26 @@ export default class PanoViewer extends Component {
...
-
-			this._photoSphereRenderer
-				.bindTexture()
-				.then(() => this._resume())
-				.catch(() => {
-					this._triggerEvent(EVENTS.ERROR, {
-						type: ERROR_TYPE.FAIL_BIND_TEXTURE,
-						message: "failed to bind texture"
-					});
+	_bindTexture() {
```
[the remark this answers is not in the document]
Comment (happyhj) [reply src/PanoViewer/PanoViewer.js]: Thanks for reporting. I'll remove that.

File: src/PanoViewer/PanoViewer.js (hunk 2 above)
Comment (happyhj) [reply src/PanoViewer/PanoViewer.js]: I agree with you. will change _activateInteraction/_inactivateInteraction to _activate/_deactivate pair.

**Pull request merged**